Source/WebCore/page/writing-tools/WritingToolsController.mm

```diff
@@ -384,6 +384,17 @@
         return;
     }
 
+    if (!state->hasReceivedText && session.compositionType == WritingTools::Session::CompositionType::SmartReply) {
+        // UIKit inserts a space prior to `willBegin` and before `didReceiveText`, so the initial session range
+        // becomes invalid and must be re-computed.
+        //
+        // FIXME: Remove this special-case logic if/when UIKit no longer injects a space.
+        if (auto selectedRange = document->selection().selection().firstRange())
```

Comment:
Does it make sense to use the selection? Aren't we supposed to not have the selection active during rewrites?

Reply:
for smart replies, it makes sense, and was the existing behavior anyways; this just makes sure it uses the up to date selection

Reply:
Should we restrict to iOS only?

Reply:
it doesn't technically matter, since it basically won't do anything anyways for macOS. I can still add the restriction though if you want?

Reply:
Would be nice to file that UIKit radar and drop it here too.